nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -943,31 +943,31 @@ private static com.sun.tools.javac.util.List<Type> convertTreesToTypes(
   /**
    * Substitutes the type arguments from a generic method invocation into the method's type.
    *
-   * @param methodInvocationTree the method invocation tree
+   * @param tree the method invocation tree
    * @param methodSymbol symbol for the invoked generic method
    * @param state the visitor state
    * @param config the NullAway config
    * @return the substituted method type for the generic method
    */
   private Type substituteTypeArgsInGenericMethodType(
-      MethodInvocationTree methodInvocationTree,
-      Symbol.MethodSymbol methodSymbol,
-      VisitorState state,
-      Config config) {
+      Tree tree, Symbol.MethodSymbol methodSymbol, VisitorState state, Config config) {
 
-    List<? extends Tree> typeArgumentTrees = methodInvocationTree.getTypeArguments();
+    List<? extends Tree> typeArgumentTrees =
+        (tree instanceof MethodInvocationTree)
+            ? ((MethodInvocationTree) tree).getTypeArguments()
+            : ((NewClassTree) tree).getTypeArguments();
     com.sun.tools.javac.util.List<Type> explicitTypeArgs = convertTreesToTypes(typeArgumentTrees);
 
     Type.ForAll forAllType = (Type.ForAll) methodSymbol.type;
     Type.MethodType underlyingMethodType = (Type.MethodType) forAllType.qtype;
 
     // There are no explicit type arguments, so use the inferred types
     if (explicitTypeArgs.isEmpty()) {
-      if (inferredSubstitutionsForGenericMethodCalls.containsKey(methodInvocationTree)) {
+      if (inferredSubstitutionsForGenericMethodCalls.containsKey(tree)) {
```

Comment:
We should change the key type of `inferredSubstitutionsForGenericMethodCalls` to be `Tree` to allow for `NewClassTree` to be a key